types/schema/data/Applicant.ts

```diff
@@ -68,18 +64,21 @@ export interface Ownership {
    */
   ownersKnown?: 'all' | 'some' | 'none';
   /**
-   * @description Names and addresses of all known owners and agricultural tenants
+   * @description Names and addresses of all known owners and agricultural tenants, including date of notice or reason requisite notice has not been given, if applicable
    */
   owners?: {
+    interest?: 'owner' | 'tenant' | 'occupier' | 'other';
     name: string;
     address: Address | string;
+    noticeGiven?: boolean;
     noticeDate?: Date;
+    noNoticeReason?: string;
```

Comment:
From the PR description it sounds like it should be that if `noticeGiven === true`, we don't need a `noNoticeReason`, and if `noticeGiven === false`, then `noNoticeReason` is required.

We can achieve this with a discriminated union of these types which I _think_ more accurately describes what's expected.

JSON Schema then handles this by using an `anyOf`

```ts
// Owners is one or the other
type Owners = OwnersNoticeGiven | OwnersNoNoticeGiven;

// Put the shared properties in one place
interface BaseOwners { 
  interest?: 'owner' | 'tenant' | 'occupier' | 'other';
  name: string;
  address: Address | string;
}

// Extend with the "discriminating" key
interface OwnersNoticeGiven extends BaseOwners {
  noticeGiven: true;
}

// Use difference in discriminator key `noticeGiven` to add additional properties
interface OwnersNoNoticeGiven extends BaseOwners {
  noticeGiven: false;
  noNoticeReason: string;
}

// Valid examples
const testNoticeNotGiven: Owners = {
  interest: 'owner',
  name: 'my name',
  address: 'my address',
  noticeGiven: false,
  noNoticeReason: 'some reason',
};

const testNoticeGiven: Owners = {
  interest: 'owner',
  name: 'my name',
  address: 'my address',
  noticeGiven: true,
};
```

Reply:
Yep - this is a smart improvement & makes expectactions a bit more clear here - updated!